pkg/kubelet/dockershim/docker_sandbox.go

```diff
@@ -165,7 +165,7 @@ func (ds *dockerService) RunPodSandbox(ctx context.Context, r *runtimeapi.RunPod
 	// on the host as well, to satisfy parts of the pod spec that aren't
 	// recognized by the CNI standard yet.
 	cID := kubecontainer.BuildContainerID(runtimeName, createResp.ID)
-	err = ds.network.SetUpPod(config.GetMetadata().Namespace, config.GetMetadata().Name, cID, config.Annotations)
+	err = ds.network.SetUpPod(config.GetMetadata().Namespace, config.GetMetadata().Name, cID, config.Annotations, config.GetDnsConfig())
```

Comment:
if you search up to L144, it looks like it's also doing dns configuration (for Unix). Is it needed for windows? 

Also, constantly changing interface to adapt to different implementation is a concern. I don't have a better suggestion for this PR, but it might be something we need to address in longer term.

Reply:
That doesn't work for Windows containers, since resolv file is not used on Windows.